### ISSUE TITLE
bug: Stencil build generates local import

### ISSUE THREAD

**Issue by mlechler** [ionic-team/stencil#4609]
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/stencil/blob/main/CONTRIBUTING.md).
- [X] I agree to follow the [Code of Conduct](https://github.com/ionic-team/stencil/blob/main/CODE_OF_CONDUCT.md).
- [x] I have searched for [existing issues](https://github.com/ionic-team/stencil/issues) that already report this problem, without success.

### Stencil Version

4.0.1

### Current Behavior

When I'm running a stencil build I get the following import in my docs.d.ts
`import { ComponentCompilerEventComplexType, ComponentCompilerMethodComplexType, ComponentCompilerPropertyComplexType, ComponentCompilerReferencedType } from './stencil-private';`

Is this correct? The file is not found, cause it is obviously not there...

### Expected Behavior

Import from the package, e.g. @stencil/core/internal

### System Info

```shell
System: node 18.16.1
    Platform: darwin (21.6.0)
   CPU Model: Apple M1 Pro (10 cpus)
    Compiler: node_modules/@stencil/core/compiler/stencil.js
       Build: 1687953951
     Stencil: 4.0.1 ⛹
  TypeScript: 5.0.4
      Rollup: 2.42.3
      Parse5: 7.1.2
      Sizzle: 2.42.3
      Terser: 5.18.1
```

### Steps to Reproduce

Run following command
`stencil build`

### Code Reproduction URL

https://github.com/mlechler/stencil-build-bug

**Comment by rwaskiewicz** [comment]
Hey @mlechler 👋 

Can you please create a code reproduction case for the team to take a look at? It's immensely helpful in understanding the issue that you're running into. Thanks!

**Comment by mlechler** [comment]
Hey @rwaskiewicz,

I added a repository url in the issue.
Simply running `npm run build` will produce the error in the `dist/docs.d.ts`.

Maybe docs-json is the problem? I guess the d.ts file is generate because of the json?

**Comment by alicewriteswrongs** [comment]
Hey @mlechler, I just published a dev build of Stencil which includes a fix for this issue. If you have time to install and try it out we can confirm the fix.

You can do so like this:

```sh
npm i @stencil/core@4.0.2-dev.1690405632.fa8e0c7 
```

**Comment by mlechler** [comment]
Hey @alicewriteswrongs, works properly I guess 👍 

Thank you for this fast solution ❤️

**Comment by alicewriteswrongs** [comment]
No problem, and glad it works! Thanks for reporting the issue! 🙂

**Comment by tanner-reits** [comment]
This was resolved in [Stencil v4.0.3](https://github.com/ionic-team/stencil/releases/tag/v4.0.3). Please re-open this issue if you are still experiencing issues after updating.
